routes/comments.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const auth = require('../modules/auth.js');
+const datastore = require('../modules/datastore.js');
+
+const COMMENT_KIND = 'Comment';
+
+/**
+ * Fetches a comment given the same comment id and pathname.
+ * If none exist, null is returned instead.
+ * Referer is used so that each page could have a separate comment thread.
+ **/
+async function getComment(commentObject) {
+  const query = datastore
+    .createQuery(COMMENT_KIND)
+    .filter('id', '=', commentObject.id)
+    .filter('pathname', '=', commentObject.pathname);
+
+  const [comments] = await datastore.runQuery(query);
+
+  if (comments.length == 0) {
+    return null;
+  } else {
+    return comments[0];
+  }
+}
+
+module.exports = function(app) {
+  /**
+   * Handles comment submission, including both new comments
+   * and updates to old comments.
+   **/
+  app.post('/comments', async function(request, response) {
+    const userObject = await auth.getUser(request.cookies.token);
+    const comment = request.body;
```

Comment:
so that this property isn't stored as true for all comments, and we reassign it anyway
```suggestion
    const comment = request.body;
    delete comment.created_by_current_user;
```

Reply:
Good point!
I originally left it as it is thinking that it's better not to waste cycle on deleting something that is going to be re-assigned anyways.

In hindsight though, deleting it does help disambiguate the entity structure.